subsys/net/l2/wifi/wifi_nm.c

```diff
@@ -115,6 +115,12 @@ int wifi_nm_register_mgd_type_iface(struct wifi_nm_instance *nm,
 
 	k_mutex_lock(&wifi_nm_lock, K_FOREVER);
 	for (int i = 0; i < CONFIG_WIFI_NM_MAX_MANAGED_INTERFACES; i++) {
+		if ((nm->mgd_ifaces[i].iface == iface)
```

Comment:
you can extend `wifi_nm_get_instance_iface` and use that here.

Reply:
Sorry, I am not sure how to extend `wifi_nm_get_instance_iface`, as we should check both `iface` and `type` here.
Besides this, `wifi_nm_get_instance_iface` is also called by `get_wifi_api`, don't want to introducing more risk here.

Reply:
> Sorry, I am not sure how to extend wifi_nm_get_instance_iface, as we should check both iface and type here.

Only for the iface check, so, no type needed. Anyways, this is fine just avoid duplication.